src/main/java/org/craftercms/studio/controller/rest/v2/ContentController.java

```diff
@@ -280,34 +289,94 @@ public ResponseBody getItemById(@RequestParam(value = REQUEST_PARAM_SITEID, requ
+                            " site '%s'", siteId));
+            result.setResponse(apiResponse);
+            responseBody.setResult(result);
+            httpServletResponse.setStatus(HttpStatus.NOT_FOUND.value());
+        } else if (paths.size() != sandboxItems.size()) {
+            List<String> found = sandboxItems.stream().map(SandboxItem::getPath).collect(Collectors.toList());
+            if (preferContent) {
+                found.addAll(sandboxItems.stream().map(si -> StringUtils.replace(si.getPath(),
+                        FILE_SEPARATOR + INDEX_FILE, "")).collect(Collectors.toList()));
+            }
+            Collection<String> missing = CollectionUtils.subtract(paths, found);
+            String missingPaths = missing.stream().collect(Collectors.joining(", "));
+            Result result = new Result();
+            ApiResponse apiResponse = new ApiResponse(CONTENT_NOT_FOUND);;
+            apiResponse.setRemedialAction(
+                    String.format("Following content paths [%s] were not found. Check that they are correct and it " +
+                            "exist in site '%s'", missingPaths, siteId));
+            result.setResponse(apiResponse);
+            responseBody.setResult(result);
+            httpServletResponse.setStatus(HttpStatus.NOT_FOUND.value());
+        } else {
+            ResultList<SandboxItem> result = new ResultList<SandboxItem>();
+            result.setEntities(RESULT_KEY_ITEMS, sandboxItems);
+            result.setResponse(OK);
+            responseBody.setResult(result);
+        }
         return responseBody;
     }
 
     @PostMapping(value = SANDBOX_ITEMS_BY_ID, produces = APPLICATION_JSON_VALUE)
-    public ResponseBody getSandboxItemsById(@RequestBody @Valid GetSandboxItemsByIdRequestBody request)
+    public ResponseBody getSandboxItemsById(@RequestBody @Valid GetSandboxItemsByIdRequestBody request,
+                                            HttpServletResponse httpServletResponse)
             throws ServiceLayerException, UserNotFoundException {
         if (!siteService.exists(request.getSiteId())) {
             throw new SiteNotFoundException(request.getSiteId());
         }
-        List<SandboxItem> sandboxItems = contentService.getSandboxItemsById(
-                request.getSiteId(), request.getIds(), request.isPreferContent());
+        List<Long> ids = request.getIds();
+        String siteId = request.getSiteId();
+        boolean preferContent = request.isPreferContent();
+        List<SandboxItem> sandboxItems = contentService.getSandboxItemsById(siteId, ids, preferContent);
+
         ResponseBody responseBody = new ResponseBody();
-        ResultList<SandboxItem> result = new ResultList<SandboxItem>();
-        result.setEntities(RESULT_KEY_ITEMS, sandboxItems);
-        result.setResponse(OK);
-        responseBody.setResult(result);
+        if (CollectionUtils.isEmpty(sandboxItems)) {
```

Comment:
This seems to be the same code as above, factor out into a common method and all it from both places.

Reply:
The algorithm is the same but actual variables and messages are different. It can not be refactored because of different data types